### PR TITLE
[ADF-2297] App-list - The component shows "No Apps found" until all the apps are loaded

### PR DESCRIPTION
**Pull request by madhukar23**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-2297

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by eromano** [comment]
can we add a customisable no app template as in the other components?

File: lib/process-services/app-list/apps-list.component.spec.ts [hunk 1]
```diff
@@ -70,6 +70,19 @@ describe('AppsListComponent', () => {
...
+    it('loading should be false by default', () => {
+        expect(component.loading).toBeFalsy();
+    });
+
+    it('should show the loading spinner when the apps are loading', () => {
+        component.loading = true;
+        fixture.detectChanges();
+        fixture.whenStable().then(() => {
+        let loadingSpinner = fixture.debugElement.query(By.css('.adf-app-list-spinner'));
+        expect(loadingSpinner.nativeElement).not.toBeNull();
```
Comment (mauriziovitale) [inline lib/process-services/app-list/apps-list.component.spec.ts]: can we check the element is exactly mat-spinner

File: lib/process-services/app-list/apps-list.component.spec.ts [hunk 2]
```diff
@@ -70,6 +70,19 @@ describe('AppsListComponent', () => {
         expect(getAppsSpy).toHaveBeenCalled();
     });
 
+    it('loading should be false by default', () => {
+        expect(component.loading).toBeFalsy();
+    });
+
+    it('should show the loading spinner when the apps are loading', () => {
```
Comment (eromano) [inline lib/process-services/app-list/apps-list.component.spec.ts]: async missing

File: lib/process-services/app-list/apps-list.component.ts [hunk 3]
```diff
@@ -40,6 +40,10 @@ export class AppsListComponent implements OnInit {
     /** (**required**) Defines the layout of the apps. There are two possible
```
Comment (eromano) [inline lib/process-services/app-list/apps-list.component.ts]: move the comment down on his property

**Pull request merged**